test/framework/src/main/java/org/elasticsearch/test/hamcrest/ElasticsearchAssertions.java

```diff
@@ -255,38 +247,27 @@ public static void assertSearchHits(SearchResponse searchResponse, String... ids
     }
 
     public static void assertSearchHitsWithoutFailures(SearchRequestBuilder requestBuilder, String... ids) {
-        var res = requestBuilder.get();
-        try {
+        assertResponse(requestBuilder, res -> {
             assertNoFailures(res);
             assertHitCount(res, ids.length);
             assertSearchHits(res, ids);
-        } finally {
-            res.decRef();
-        }
+        });
     }
 
     public static void assertSortValues(SearchRequestBuilder searchRequestBuilder, Object[]... sortValues) {
-        var searchResponse = searchRequestBuilder.get();
-        try {
-            assertNoFailures(searchResponse);
-            SearchHit[] hits = searchResponse.getHits().getHits();
+        assertResponse(searchRequestBuilder, res -> {
```

Comment:
I guess this can be `assertNoFailuresAndResponse`?

Reply:
done